src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -292,6 +300,10 @@ public function getEmbeddedAttributes()
             }
         }
 
+        if ($this->embeddedAttributesFilter !== null) {
```

Comment:
One thing I wonder though, `ProductLazyArray` already has internal management of the whitelist, so using this filter class is a bit redundant The real question is more about knowing why the whitelist from `ProductLazyArray` and `EmbeddedAttributesFilter` are not consistent?
Which one is the truth?

Reply:
The built-in whitelist of ProductLazyArray is not meant to be extended, it's a static `protected` function call. You need to create a child class to extend it.

The EmbeddedAttributesFilter class built by @eternoendless is meant to be extended so I would rather use it.

Reply:
Ok, this much makes sense, but there is still lot of difference between the two, `ProductLazyArray` has much more whitelist fields than the filter, so this change will remove a lot of fields that were initially present (which is a kind of BC break) So the technical solution seems correct, but we also need to think about which fields need to be filtered or not based on the two lists here

And if we end up using EmbeddedAttributesFilter, I guess the internal mechanism in ProductLazyArray is not useful anymore and could be completely replaced

Reply:
> has much more whitelist fields than the filter, so this change will remove a lot of fields that were initially present 

This is exactly the purpose of this PR 😄 . The whitelist inside ProductLazyArray contain data such as SQL IDs 🤔  I don't understand why they are exposed.

> but we also need to think about which fields need to be filtered or not based on the two lists here

I think the current target of EmbeddedAttributesFilter is quite good enough, and can be extended by modules to add missing items.

> I guess the internal mechanism in ProductLazyArray is not useful anymore and could be completely replaced

Agreed

Reply:
See https://github.com/PrestaShop/PrestaShop/pull/25755#issuecomment-910444544

Reply:
How can this value be `null` as you define it in the constructor ?

Reply:
Looks like we missed the intent of EmbeddedProductAttributs, which I traced back to #7027

So basically LazyArrays aren't meant to be filtered in the first place because they are supposed to provide data to templates, and templates need to access all data to do their stuff. It's not necessarily a problem to provide them with more than they need, because information doesn't naturally leak out. Conversely, data being sent to the `prestashop` js object though is to be filtered via whitelist to avoid leaking information. We didn't realize that the `embedded_attributes` key was specifically added as a filtered out list of product attributes to be published as a data attribute, so it would be appropriate to filter it using `EmbeddedAttributesFilter`.

Reply:
As @Progi1984  saide, the `if` is not needed because `$this->embeddedAttributesFilter` is never null.